src/main/java/com/compilercharisma/chameleonbusinessstudio/controller/AppointmentController.java

```diff
@@ -121,7 +122,27 @@ public ResponseEntity update(@RequestBody AppointmentEntity appointment){
             appointments.updateAppointment(appointment);
         }
         
-        return ResponseEntity.status(HttpStatus.ACCEPTED).build();
+        return Mono.just(ResponseEntity.status(HttpStatus.ACCEPTED).build());
+    }
+
+    /**
+     * Books the currently logged in user for the given appointment.
+     * @param appointment the appointment to book the logged-in user for.
+     * @return either a bad request or an OK response containing the updated 
+     *  appointment
+     */
+    @PostMapping("/book-me")
+    public Mono<ResponseEntity<AppointmentEntity>> bookMe(@RequestBody AppointmentEntity appointment){
+        if(!appointments.isAppointmentValid(appointment)){
+            return Mono.just(ResponseEntity.badRequest().body(appointment));
```

Comment:
Suggestion: Is it possible to start console.log()`ing a description for instances where a bad request is being returned? Might make it easier to debug an error 400/500 for example.

Reply:
Just realized "console.log()" is TS syntax, but I mean it's Java equivalent lol

Reply:
Good idea! That will make debugging a lot easier